core/src/main/scala/org/apache/spark/scheduler/CookSchedulerContext.scala

```diff
@@ -109,6 +117,21 @@ case class CookSchedulerContext(
         tgz.split(",").map(_.trim).toList
       }
 
+  /**
+    * Parse the Cook constraints from configuration.
+    *
+    * The Cook constraint(s) could be provided by the following Spark
+    * configuration.
+    *  - spark.cook.job.constraint=[foo,EQUALS,bar]
+    *  - spark.cook.job.constraint.1=[foo1,EQUALS,bar1]
+    *  - spark.cook.job.constraint.2=[foo2,EQUALS,bar2]
+    */
+  val cookJobConstraints: Seq[Constraint] = conf.getAll.filter { case (key, _) =>
+    key.matches(SPARK_COOK_JOB_CONSTRAINT + "(\\.(\\d)+$|$)")
```

Comment:
My regex foo is pretty weak so I don't understand the $|$ at the end. What is that suppose to be doing and can you simplify it?

Reply:
`|` means "or"

So that is either "spark.cook.job.constraint.0=[a=b]" or simply "spark.cook.job.constraint=[a=b]"